server/routes/imageUpload.js

```diff
@@ -0,0 +1,27 @@
+const express = require("express");
+const { validate } = require("../middlewares/validate");
+const router = express.Router();
+const upload = require("../services/imageUpload");
+const uploadFile = upload.single("image");
+
+const uploadImage = (req, res) => {
+  uploadFile(req, res, (err) => {
+    if (err) {
+      return res.json({
+        success: false,
+        errors: {
+          title: "Image Upload Error",
+          detail: err.message,
+          error: err,
+        },
+      });
+    } else {
+      const imageUrl = req.file.location;
+      res.status(200).json({ imageUrl: imageUrl });
+    }
+  });
+};
+
+router.post("/", validate, uploadImage);
```

Comment:
not too important, but if you're already making changes, I'm pretty sure validate is not necessary here since there aren't any prior checks being made.

Reply:
oh, yes you are right, since we do have the authMiddleWare in use we don't necessarily need this one.